src/helm/manifests/ralph-env-secret.yaml

```diff
@@ -0,0 +1,139 @@
+  # RALPH_BACKENDS__DATA__S3__SESSION_TOKEN: ""
+  # RALPH_BACKENDS__DATA__S3__ENDPOINT_URL: ""
+  # RALPH_BACKENDS__DATA__S3__DEFAULT_REGION: ""
+  # RALPH_BACKENDS__DATA__S3__DEFAULT_BUCKET_NAME: ""
+  # RALPH_BACKENDS__DATA__S3__LOCALE_ENCODING: ""
+  # RALPH_BACKENDS__DATA__S3__READ_CHUNK_SIZE: ""
+  # RALPH_BACKENDS__DATA__S3__WRITE_CHUNK_SIZE: ""
+
+  ## Elasticsearch data backend
+  # RALPH_BACKENDS__DATA__ES__HOSTS: "http://elasticsearch:9200"
+  # RALPH_BACKENDS__DATA__ES__DEFAULT_INDEX: "statements"
+  # RALPH_BACKENDS__DATA__ES__ALLOW_YELLOW_STATUS: "False"
+  # RALPH_BACKENDS__DATA__ES__CLIENT_OPTIONS__ca_certs: "False"
+  # RALPH_BACKENDS__DATA__ES__CLIENT_OPTIONS__verify_certs: "False"
+  # RALPH_BACKENDS__DATA__ES__LOCALE_ENCODING: "utf8"
+  # RALPH_BACKENDS__DATA__ES__POINT_IN_TIME_KEEP_ALIVE: "1m"
+  # RALPH_BACKENDS__DATA__ES__READ_CHUNK_SIZE: "500"
+  # RALPH_BACKENDS__DATA__ES__REFRESH_AFTER_WRITE: "False"
+  # RALPH_BACKENDS__DATA__ES__WRITE_CHUNK_SIZE: "500"
+
+  ## Mongo data backend
+  # RALPH_BACKENDS__DATA__MONGO__CONNECTION_URI: "mongodb://mongo:27017/"
+  # RALPH_BACKENDS__DATA__MONGO__DEFAULT_COLLECTION: "foo"
+  # RALPH_BACKENDS__DATA__MONGO__DEFAULT_DATABASE: "statements"
+  # RALPH_BACKENDS__DATA__MONGO__CLIENT_OPTIONS__document_class: ""
+  # RALPH_BACKENDS__DATA__MONGO__CLIENT_OPTIONS__tz_aware: "False"
+  # RALPH_BACKENDS__DATA__MONGO__LOCALE_ENCODING: "utf8"
+  # RALPH_BACKENDS__DATA__MONGO__READ_CHUNK_SIZE: "500"
+  # RALPH_BACKENDS__DATA__MONGO__WRITE_CHUNK_SIZE: "500"
+
+  ## ClickHouse data backend
+  # RALPH_BACKENDS__DATA__CLICKHOUSE__HOST: "clickhouse"
+  # RALPH_BACKENDS__DATA__CLICKHOUSE__PORT: "8123"
+  # RALPH_BACKENDS__DATA__CLICKHOUSE__DATABASE: "xapi"
+  # RALPH_BACKENDS__DATA__CLICKHOUSE__EVENT_TABLE_NAME: "xapi_events_all"
+  # RALPH_BACKENDS__DATA__CLICKHOUSE__USERNAME: ""
+  # RALPH_BACKENDS__DATA__CLICKHOUSE__PASSWORD: ""
+  # RALPH_BACKENDS__DATA__CLICKHOUSE__CLIENT_OPTIONS__date_time_input_format: ""
+  # RALPH_BACKENDS__DATA__CLICKHOUSE__LOCALE_ENCODING: "utf8"
+  # RALPH_BACKENDS__DATA__CLICKHOUSE__READ_CHUNK_SIZE: "500"
+  # RALPH_BACKENDS__DATA__CLICKHOUSE__WRITE_CHUNK_SIZE: "500"
+
+  ## FS data backend
+  # RALPH_BACKENDS__DATA__FS__DEFAULT_DIRECTORY_PATH: "."
+  # RALPH_BACKENDS__DATA__FS__DEFAULT_QUERY_STRING: "*"
+  # RALPH_BACKENDS__DATA__FS__LOCALE_ENCODING: "utf8"
+  # RALPH_BACKENDS__DATA__FS__READ_CHUNK_SIZE: "4096"
+  # RALPH_BACKENDS__DATA__FS__WRITE_CHUNK_SIZE: "4096"
+
+  ## LRS HTTP backend
+  # RALPH_BACKENDS__DATA__LRS__BASE_URL: "http://ralph:secret@0.0.0.0:8100/"
+  # RALPH_BACKENDS__DATA__LRS__USERNAME: "ralph"
+  # RALPH_BACKENDS__DATA__LRS__PASSWORD: "secret"
+  # RALPH_BACKENDS__DATA__LRS__HEADERS__X_EXPERIENCE_API_VERSION: "1.0.3"
+  # RALPH_BACKENDS__DATA__LRS__HEADERS__CONTENT_TYPE: "application/json"
+  # RALPH_BACKENDS__DATA__LRS__STATUS_ENDPOINT: "/__heartbeat__"
+  # RALPH_BACKENDS__DATA__LRS__STATEMENTS_ENDPOINT: "/xAPI/statements"
+---
```

Comment:
For me except the key ending by ``__USERNAME``,  ``__PASSWORD``, ``__ACCESS_KEY_ID`` or ``__SECRET_ACCESS_KEY`` should be in a configmap.

Reply:
Good remark. I created a ConfigMap manifest for non-secret environment variables.
I really don't want to add them as variable in the Helm chart :grin: